src/common/tree_node_serializer.py

```diff
@@ -178,8 +178,13 @@ def tree_node_from_dict(
                     content_attribute.attribute_type = child["type"]
                     list_child_attribute = content_attribute
 
+                if "ref" in child:
+                    child_uid = child["ref"]
+                else:
+                    child_uid = child.get("_id", "")
+
```

Comment:
Should this rather do a check on type, and just assume "ref" is there if it's a "reference type"?

Reply:
sure!
that would make it a bit easier to read